src/events/cd-user-ticket-list-item.vue

```diff
@@ -0,0 +1,295 @@
+    </div>
+  </div>
+</template>
+<script>
+  import moment from 'moment';
+  import { sortBy, pick } from 'lodash';
+  import cdDateFormatter from '@/common/filters/cd-date-formatter';
+  import cdTimeFormatter from '@/common/filters/cd-time-formatter';
+  import cdUrlFormatter from '@/common/filters/cd-url-formatter';
+  import cdAttendee from './cd-attendee';
+  import EventService from './service';
+  import EventsUtil from './util';
+
+  export default {
+    name: 'user-ticket-list-item',
+    props: ['event', 'dojo', 'usersDojos', 'users'],
+    data() {
+      return {
+        applications: [],
+        sessions: {},
+        tickets: {},
+      };
+    },
+    computed: {
+      hasApplications() {
+        return this.applications.length > 0;
+      },
+      isFull() {
+        let totalEventCapacity = 0;
+        let totalTicketsBooked = 0;
+        this.event.sessions.forEach((session) => {
+          session.tickets.forEach((ticket) => {
+            totalEventCapacity += ticket.quantity;
+            totalTicketsBooked += ticket.approvedApplications;
+          });
+        });
+        return totalEventCapacity === totalTicketsBooked;
+      },
+      nextStartTime() {
+        return EventsUtil.getNextStartTime(this.event);
+      },
+      isRecurring() {
+        return EventsUtil.isRecurring(this.event);
+      },
+      recurringFrequencyInfo() {
+        return EventsUtil.buildRecurringFrequencyInfo(this.event);
+      },
+      isPastEvent() {
+        const now = moment();
+        const eventStartTime = moment(this.event.dates[this.event.dates.length - 1].startTime);
+        eventStartTime.subtract(eventStartTime.utcOffset(), 'minutes');
+        return now.isAfter(eventStartTime);
+      },
+      bookLink() {
+        return `/dojo/${this.dojo.id}/event/${this.event.id}`;
+      },
+      formattedStartTime() {
+        return this.$options.filters.cdTimeFormatter(this.event.dates[0].startTime);
+      },
+      formattedEndTime() {
+        return this.$options.filters.cdTimeFormatter(this.event.dates[0].endTime);
+      },
+      formattedFirstDate() {
+        const sortedDates = sortBy(this.event.dates, date => date.startTime);
+        return this.$options.filters.cdDateFormatter(sortedDates[0].startTime);
+      },
+      formattedLastDate() {
+        const sortedDates = sortBy(this.event.dates, date => date.startTime);
+        return this.$options.filters.cdDateFormatter(sortedDates[sortedDates.length - 1].startTime);
+      },
+    },
+    methods: {
+      prepareSessions() {
+        this.event.sessions.forEach((s) => {
+          this.sessions[s.id] = s;
+        });
+      },
+      prepareTickets() {
+        this.event.sessions.forEach((s) => {
+          s.tickets.forEach((t) => {
+            this.tickets[t.id] = t;
+          }, this);
+        });
+      },
+      async cancel() {
+        const payload = [];
+        const commonFields = ['dojoId', 'eventId', 'sessionId', 'ticketName',
```

Comment:
Probably move out of the component so we're not initialising every time we call cancel?

Reply:
fair